### PR TITLE
Stellar payments listening for identity bridge service

### PR DESCRIPTION
**Pull request by justwudi**
# New packages
- Add `services/identity/db` package for `Database` interface/implementations and model structs.
- Add `services/identity/listener` package for listeners such as:
  - Stellar payments streaming
  - Worker job to create link requests based on incoming stellar payments (not implemented)
  - Ethereum smart contract events (not implemented)
- Add `services/identity/eventhandler` package for event handlers such as:
  - Incoming stellar payments
  - New link requests (not implemented)
  - New ethereum smart contract events (not implemented)
- Add `support/zapencoder` package, which is a modified version of `zap`'s encoder for pretty printing.
- Add `support/stellar` package for stellar utilities such as the  extraction and validation of ethereum address from stellar transaction memo.

# Features
- Listen to stellar events, record into database.
- Validates the ethereum address from the transaction memo, stores the status of the validation.

### CODE REVIEW

File: packages/demo/bridge/services/identity/db/main.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	LastCreated  string `db:"last_created"`
+	LastModified string `db:"last_modified"`
+}
+
+type StellarAccount struct {
+	ID      int64       `db:"id"`
+	Address string      `db:"address"`
+	Cursor  null.String `db:"cursor"`
+	Status  int64       `db:"status"`
+	Timestamp
```
Comment (imwaihon) [inline packages/demo/bridge/services/identity/db/main.go:51]: This timestamp is server sided? Is there a Tx timestamp?

File: packages/demo/bridge/services/identity/db/postgres.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		InsertInto(tableStellarAccounts).
+		Columns(columnAddress).
+		Values(address).
+		Returning(columnID).
+		Load(&id)
+	if err != nil {
+		return nil, err
+	}
+
+	err = tx.
```
Comment (imwaihon) [inline packages/demo/bridge/services/identity/db/postgres.go:46]: Is there no way to get res from a successful insert (no error) rather then querying the exact same thing again?

File: packages/demo/bridge/services/identity/eventhandler/stellar_payment_handler.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	if id, err := strconv.Atoi(payment.ID); err == nil {
+		receivedPayment.PaymentID = int64(id)
+	}
+
+	switch payment.Type {
+	case "create_account":
+		receivedPayment.FromAccount = payment.Funder
+	}
+
+	if !memo.Valid {
```
Comment (imwaihon) [inline packages/demo/bridge/services/identity/eventhandler/stellar_payment_handler.go:58]: Is there a check on valid ethereum address? based on checksum

File: packages/demo/bridge/support/stellar/main.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+func ExtractMemoToEthereumAddress(memoType, memoValue string) (ethAddress string, err error) {
+	switch memoType {
+	case "hash":
+		memoBytes, err := base64.StdEncoding.DecodeString(memoValue)
+		if err != nil {
+			return "", err
+		}
+		if len(memoBytes) != 32 {
+			return "", ErrInvalidEthAddress
```
Comment (imwaihon) [inline packages/demo/bridge/support/stellar/main.go:43]: https://ethereum.stackexchange.com/questions/1374/how-can-i-check-if-an-ethereum-address-is-valid

does this cover the case of checksummed address?

**Review by imwaihon — CHANGES_REQUESTED** [review]
For go packages do we have to commit all the code? Or just because this is demo.

File: packages/demo/bridge/services/identity/db/main.go (hunk 1 above)
Comment (justwudi) [reply packages/demo/bridge/services/identity/db/main.go:51]: This timestamp is server sided. I'm not storing the tx timestamp of the stellar events.

File: packages/demo/bridge/services/identity/db/postgres.go (hunk 2 above)
Comment (justwudi) [reply packages/demo/bridge/services/identity/db/postgres.go:46]: It's possible with raw sql, we'll lose the syntactic sugar provided by the `dbr` library though. Inserting a new account in this `LoadStellarAccount` function is not likely to happen often, so there won't be any performance issues if we do a insert followed by a select.

File: packages/demo/bridge/services/identity/eventhandler/stellar_payment_handler.go (hunk 3 above)
Comment (justwudi) [reply packages/demo/bridge/services/identity/eventhandler/stellar_payment_handler.go:58]: No, ethereum address will only have checksum if they are represented in **string** form with capitalization. The memo that we are expecting here is a hexadecimal in bytes form.

File: packages/demo/bridge/support/stellar/main.go (hunk 4 above)
Comment (justwudi) [reply packages/demo/bridge/support/stellar/main.go:43]: No, the `memoBytes` here is not a string with capitalization. Stellar `MEMO_TEXT` type memo field only allows a maximum of 28 bytes, we won't be able to fit a 40 character address string into it.

**Review by imwaihon — APPROVED** [review]

**Pull request merged**